File: src/main/java/com/anatawa12/crashReportPublisher/CrashReportPublisherTweaker.java
package com.anatawa12.crashReportPublisher;

import net.minecraft.launchwrapper.ITweaker;
import net.minecraft.launchwrapper.LaunchClassLoader;

import java.io.File;
import java.util.List;

@SuppressWarnings("unused")
public class CrashReportPublisherTweaker implements ITweaker {
    @Override
    public void acceptOptions(List<String> args, File gameDir, File assetsDir, String profile) {
        if (gameDir == null) gameDir = new File(".");
        CrashReportPublisher.loadConfig(gameDir);
    }

    @Override
    public void injectIntoClassLoader(LaunchClassLoader classLoader) {
        classLoader.registerTransformer(CrashReportPublisherClassTransformer.class.getName());
    }

    @Override
    public String getLaunchTarget() {
        return null;
    }

    @Override
    public String[] getLaunchArguments() {
        return new String[0];
    }
}

File: src/main/kotlin/com/anatawa12/crashReportPublisher/CrashReportPublisherTweaker.kt
package com.anatawa12.crashReportPublisher

import net.minecraft.launchwrapper.ITweaker
import net.minecraft.launchwrapper.LaunchClassLoader
import java.io.File

@Suppress("unused")
class CrashReportPublisherTweaker : ITweaker {
    override fun acceptOptions(args: MutableList<String>?, gameDir: File?, assetsDir: File?, profile: String?) {
        loadConfig(gameDir ?: File("."))
    }

    override fun injectIntoClassLoader(classLoader: LaunchClassLoader) {
        classLoader.registerTransformer(CrashReportPublisherClassTransformer::class.java.name)
    }

    override fun getLaunchTarget(): String? = null

    override fun getLaunchArguments(): Array<String> = emptyArray()
}
